pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/CheckTopicIsSpecial.java

```diff
@@ -0,0 +1,36 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.broker.service.persistent;
+
+import org.apache.logging.log4j.core.util.Assert;
+import org.apache.pulsar.functions.worker.WorkerConfig;
+
+public class CheckTopicIsSpecial {
+    public static  boolean checkTopicIsFunctionWorkerService(PersistentTopic topic)  {
+        if (!Assert.isNonEmpty(topic)){
+            throw new IllegalArgumentException("topic can`t be null");
+        }
+       WorkerConfig workerConfig =  topic.getBrokerService().getPulsar().getWorkerConfig().get();
+        String topicName = topic.getName();
+        return workerConfig.getClusterCoordinationTopic().equals(topicName)
+                || workerConfig.getFunctionAssignmentTopic().equals(topicName)
+                || workerConfig.getFunctionMetadataTopic().equals(topicName);
```

Comment:
This can only work for the broker run with the function worker? If the function working running independently, here will get an incorrect worker config.

I think we already have a PR to support transaction buffer and pending ack lazy creation, if new transactions happens on a topic, the transaction buffer and the pending ack will not be created?